github/login_test.go

```diff
@@ -108,3 +109,23 @@ func TestValidateResponse(t *testing.T) {
 	assert.Equal(t, ErrUnableToGetGithubUser, validateResponse(validUser, invalidResponse, nil))
 	assert.Equal(t, ErrUnableToGetGithubUser, validateResponse(&github.User{}, validResponse, nil))
 }
+
+func Test_githubClientFromAuthURL(t *testing.T) {
+	for _, test := range []struct {
+		authURL          string
+		expClientBaseURL string
+	}{
+		{authURL: "https://github.com/login/oauth/authorize/", expClientBaseURL: "https://api.github.com/"},
+		{authURL: "https://github.com/login/oauth/authorize", expClientBaseURL: "https://api.github.com/"},
+		{authURL: "https://github.mycompany.com/login/oauth/authorize", expClientBaseURL: "https://github.mycompany.com/api/v3/"},
+		{authURL: "http://github.mycompany.com/login/oauth/authorize", expClientBaseURL: "http://github.mycompany.com/api/v3/"},
+	} {
+		client, err := githubClientFromAuthURL(test.authURL, nil)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if got, want := client.BaseURL.String(), test.expClientBaseURL; got != want {
```

Comment:
gologin already makes use of testify assert, which you can use. assert.Equal(t, client.BaseURL.String(), c.expectedBaseURL)